src/plt-mach-o.c

```diff
@@ -247,6 +255,13 @@ void plt_set_offsets(plt_offset *offset, size_t nb_off, plt_fn *newval)
     for (size_t i = 0; i < nb_off; ++i) {
         if (!offset[i].oldval)
             offset[i].oldval = *offset[i].offset;
+#ifdef MMK_ARCH_ARM64
+        const long page_size = sysconf(_SC_PAGESIZE);
+        mmk_assert(page_size > 0);
+        char *page_start = (char *) (((uintptr_t) offset[i].offset) & ~(page_size - 1));
+        mmk_assert(!mprotect (page_start, page_size, PROT_READ | PROT_WRITE));
+        __clear_cache(page_start, page_start + page_size);
+#endif
```

Comment:
@eboasson mind to leave a comment as to why this is necessary and why only for `aarch64` architectures?

Reply:
@hidmic some of the GOTs are mapped read-only on (at least) Big Sur on an M1, and so it has to be remapped read/write to do the patching (I wouldn't be surprised if Apple were to disallow remapping it as writable at some point, making it impossible to do this GOT poisoning, but that's a bridge we shouldn't try to cross until we get to it).

Clearing the cache is needed for ARM but not for Intels because of the differences in memory models; that one I simply picked up from the Linux aarch64 variant.

The reason I have restricted it to aarch64 is simply a case of trying to limit the damage I do: I haven't tested this properly on anything other than my M1-based laptop.

Reply:
Thanks for the explanation.